VoxelSniperCore/src/main/java/com/github/kevindagame/util/schematic/DataFolderSchematicReader.kt

```diff
@@ -0,0 +1,109 @@
+package com.github.kevindagame.util.schematic
+
+import com.github.kevindagame.VoxelSniper
+import net.sandrohc.schematic4j.SchematicLoader
+import net.sandrohc.schematic4j.schematic.Schematic
+import java.io.File
+
+class DataFolderSchematicReader : ISchematicReader {
+    override fun getSchematicFile(name: String): File? {
+        //check for exact name
+        val exactSchematic = VoxelSniper.voxelsniper.fileHandler.getDataFile(SCHEMATIC_FILE_ROOT_PATH + name)
```

Comment:
I think it is better to use ```new File(baseDirectory, file);``` or simular.

Reply:
I disagree. Until now we have used the file handler everywhere. Making a file directly would require getting the full path to the plugin root, which is extra work

Reply:
you don't https://github.com/KevinDaGame/VoxelSniper-Reimagined/blob/master/VoxelSniperCore/src/main/java/com/github/kevindagame/util/Messages.java#L383C92-L383C92

Reply:
Also I am not sure if using '/' is platform dependent or not..

Reply:
> you don't https://github.com/KevinDaGame/VoxelSniper-Reimagined/blob/master/VoxelSniperCore/src/main/java/com/github/kevindagame/util/Messages.java#L383C92-L383C92

That's more an issue on that part. I prefer to leave this file logic to the main file handler. It might be a good idea to refactor this in other places

Reply:
Maybe this would be a nice option, we could add/change a method in filehandler ```File getDataFile(String... path)```, which could take all path 'parts', without having to concatenate the strings, which I think is not the way it is meant to be done in Java (could be wrong here, but most examples either use File or Paths).
```    default File getDataFile(String... path) {
        File f = getDataFolder();
        for (String p : path) {
            f = new File(f, p);
        }
        return f;
    }```

Reply:
INFO: Why don't you like the way I'm using it? I'd like to stay away from the File constructor wherever possible, to prevent divergence later on.

Reply:
I do think though that the way we access the filehandler isn't nice. Perhabs this could be solved using dependency injection

Reply:
I will approve the PR, since the same issue was already present before I think, but we should indeed look into improving this.
I feel like we should, if we want to be able to change the implementation, supply every part of the path to the filehandler seperatly, so that it can traferse the filesystem in whatever way it needs to.